### ISSUE TITLE
Input config crashes on Linux

### ISSUE THREAD

**Issue by Zowayix** [StapleButter/melonDS#267]
I think this might be a different issue than #264 (I may be entirely wrong), as the backtrace I get is a bit different (unless I just don't know how to work gdb properly), but also it doesn't seem to crash until a few seconds of having the dialog open. If I'm quick enough to set A to a button on my gamepad, it'll actually go so far as to write `Joy_A=1` to melonDS.ini and read it next time I try opening the input dialog. But it'll crash with a segmentation fault before I can remap all the controls.

Aforementioned backtrace: 

```
(gdb) run
Starting program: /tmp/melonDS/build/melonDS 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
melonDS 0.7
http://melonds.kuribo64.net/
[New Thread 0x7ffff18eb700 (LWP 23628)]
[New Thread 0x7ffff10ea700 (LWP 23629)]
[New Thread 0x7fffdb212700 (LWP 23630)]
[New Thread 0x7fffdaa11700 (LWP 23631)]
[New Thread 0x7fffd9efd700 (LWP 23632)]
[New Thread 0x7fffd95fc700 (LWP 23633)]
[New Thread 0x7fffb2611700 (LWP 23637)]

Thread 1 "melonDS" received signal SIGSEGV, Segmentation fault.
0x0000000000000000 in ?? ()
(gdb) bt
#0  0x0000000000000000 in  ()
#1  0x00007ffff79aae6d in  () at /usr/lib/libgtk-3.so.0
#2  0x00007ffff79d3736 in  () at /usr/lib/libgtk-3.so.0
#3  0x00007ffff79be8ab in  () at /usr/lib/libgtk-3.so.0
#4  0x00007ffff79d3670 in  () at /usr/lib/libgtk-3.so.0
#5  0x00007ffff79c0ef6 in  () at /usr/lib/libgtk-3.so.0
#6  0x00007ffff79bfc95 in  () at /usr/lib/libgtk-3.so.0
#7  0x00007ffff79c09cd in  () at /usr/lib/libgtk-3.so.0
#8  0x00007ffff79c0a1b in  () at /usr/lib/libgtk-3.so.0
#9  0x00007ffff79c0a1b in  () at /usr/lib/libgtk-3.so.0
#10 0x00007ffff79c0a1b in  () at /usr/lib/libgtk-3.so.0
#11 0x00007ffff79c0a1b in  () at /usr/lib/libgtk-3.so.0
#12 0x00007ffff79c0a1b in  () at /usr/lib/libgtk-3.so.0
#13 0x00007ffff79c0a1b in  () at /usr/lib/libgtk-3.so.0
#14 0x00007ffff79c0a1b in  () at /usr/lib/libgtk-3.so.0
#15 0x00007ffff79c0a1b in  () at /usr/lib/libgtk-3.so.0
#16 0x00007ffff79c0a1b in  () at /usr/lib/libgtk-3.so.0
#17 0x00007ffff79a69f0 in  () at /usr/lib/libgtk-3.so.0
#18 0x00007ffff739c3d5 in g_closure_invoke () at /usr/lib/libgobject-2.0.so.0
#19 0x00007ffff7389195 in  () at /usr/lib/libgobject-2.0.so.0
#20 0x00007ffff738d01e in g_signal_emit_valist () at /usr/lib/libgobject-2.0.so.0
#21 0x00007ffff738da80 in g_signal_emit () at /usr/lib/libgobject-2.0.so.0
#22 0x00007ffff778231b in  () at /usr/lib/libgdk-3.so.0
#23 0x00007ffff776cb2b in  () at /usr/lib/libgdk-3.so.0
#24 0x00007ffff72b4b63 in  () at /usr/lib/libglib-2.0.so.0
#25 0x00007ffff72b5271 in g_main_context_dispatch () at /usr/lib/libglib-2.0.so.0
#26 0x00007ffff72b6f89 in  () at /usr/lib/libglib-2.0.so.0
#27 0x00007ffff72b7f62 in g_main_loop_run () at /usr/lib/libglib-2.0.so.0
#28 0x00007ffff7a7a1af in gtk_main () at /usr/lib/libgtk-3.so.0
#29 0x0000555555605df7 in uiMain () at /tmp/melonDS/src/libui_sdl/libui/unix/main.c:53
#30 0x000055555559f089 in main(int, char**) (argc=1, argv=0x7fffffffe1a8) at /tmp/melonDS/src/libui_sdl/main.cpp:1592
```

Version is built from git just now (ie commit fb284f3), distro is Antergos Linux, game controllers plugged in right now are an Xbox One gamepad and an N64 controller via Raphnet adapter v3.5, hopefully there's no other info that I should have included that I didn't.

**Comment by StapleButter** [comment]
ugh, I don't quite see what's exploding there

I'll have to look into it

**Comment by Zowayix** [comment]
Seems that if I disconnect both controllers, it doesn't crash. But then I can't configure them.

Hmm...

**Comment by Zowayix** [comment]
Same thing happens if I only have one of the controllers plugged in.

**Comment by Zowayix** [comment]
I've discovered how to invoke the dark magic of gdb to actually get all the threads, which is probably more helpful:

```
(gdb) thread apply all bt

Thread 8 (Thread 0x7fffb2611700 (LWP 31175)):
#0  0x0000000000000000 in  ()
#1  0x00007ffff79c4f46 in  () at /usr/lib/libgtk-3.so.0
#2  0x00007ffff79aa3f2 in  () at /usr/lib/libgtk-3.so.0
#3  0x00007ffff79d3736 in  () at /usr/lib/libgtk-3.so.0
#4  0x00007ffff79be8ab in  () at /usr/lib/libgtk-3.so.0
#5  0x00007ffff79d3670 in  () at /usr/lib/libgtk-3.so.0
#6  0x00007ffff79c0ef6 in  () at /usr/lib/libgtk-3.so.0
#7  0x00007ffff79bfc95 in  () at /usr/lib/libgtk-3.so.0
#8  0x00007ffff79bfe86 in  () at /usr/lib/libgtk-3.so.0
#9  0x00007ffff7b25d51 in  () at /usr/lib/libgtk-3.so.0
#10 0x00007ffff7bd0263 in  () at /usr/lib/libgtk-3.so.0
#11 0x00007ffff7bd04c1 in gtk_widget_realize () at /usr/lib/libgtk-3.so.0
#12 0x00007ffff7bd4f49 in gtk_widget_set_parent () at /usr/lib/libgtk-3.so.0
#13 0x00007ffff7954d4c in  () at /usr/lib/libgtk-3.so.0
#14 0x00007ffff739aa60 in g_cclosure_marshal_VOID__OBJECTv () at /usr/lib/libgobject-2.0.so.0
#15 0x00007ffff738d16e in g_signal_emit_valist () at /usr/lib/libgobject-2.0.so.0
#16 0x00007ffff738da80 in g_signal_emit () at /usr/lib/libgobject-2.0.so.0
#17 0x00007ffff79a61e6 in gtk_container_add () at /usr/lib/libgtk-3.so.0
#18 0x00007ffff796832e in  () at /usr/lib/libgtk-3.so.0
#19 0x00007ffff7968fb8 in gtk_button_set_label () at /usr/lib/libgtk-3.so.0
#20 0x00005555555fc5de in uiButtonSetText (b=0x5555562ab9c0, text=0x7fffb2610bc0 "Button 2") at /tmp/melonDS/src/libui_sdl/libui/unix/button.c:33
#21 0x00005555555a04e4 in DlgInputConfig::JoyPoll(unsigned int, void*) (interval=100, param=0x0) at /tmp/melonDS/src/libui_sdl/DlgInputConfig.cpp:172
#22 0x00007ffff71664fc in  () at /usr/lib/libSDL2-2.0.so.0
#23 0x00007ffff7165f9f in  () at /usr/lib/libSDL2-2.0.so.0
#24 0x00007ffff71e96ea in  () at /usr/lib/libSDL2-2.0.so.0
#25 0x00007ffff6127a9d in start_thread () at /usr/lib/libpthread.so.0
#26 0x00007ffff6d03b23 in clone () at /usr/lib/libc.so.6

Thread 7 (Thread 0x7fffd95e3700 (LWP 31174)):
#0  0x00007ffff6cf8d16 in ppoll () at /usr/lib/libc.so.6
#1  0x00007ffff0c61a13 in pa_mainloop_poll () at /usr/lib/libpulse.so.0
#2  0x00007ffff0c61fe0 in pa_mainloop_iterate () at /usr/lib/libpulse.so.0
#3  0x00007ffff71c7847 in  () at /usr/lib/libSDL2-2.0.so.0
#4  0x00007ffff711f046 in  () at /usr/lib/libSDL2-2.0.so.0
#5  0x00007ffff7165f9f in  () at /usr/lib/libSDL2-2.0.so.0
#6  0x00007ffff71e96ea in  () at /usr/lib/libSDL2-2.0.so.0
#7  0x00007ffff6127a9d in start_thread () at /usr/lib/libpthread.so.0
#8  0x00007ffff6d03b23 in clone () at /usr/lib/libc.so.6

Thread 6 (Thread 0x7fffd9ee4700 (LWP 31173)):
#0  0x00007ffff6cd06a8 in nanosleep () at /usr/lib/libc.so.6
#1  0x00007ffff71ebc56 in  () at /usr/lib/libSDL2-2.0.so.0
#2  0x000055555559bc28 in EmuThreadFunc(void*) (burp=0x0) at /tmp/melonDS/src/libui_sdl/main.cpp:353
#3  0x00007ffff7165f9f in  () at /usr/lib/libSDL2-2.0.so.0
#4  0x00007ffff71e96ea in  () at /usr/lib/libSDL2-2.0.so.0
#5  0x00007ffff6127a9d in start_thread () at /usr/lib/libpthread.so.0
#6  0x00007ffff6d03b23 in clone () at /usr/lib/libc.so.6

Thread 5 (Thread 0x7fffdaa11700 (LWP 31172)):
#0  0x00007ffff6cf8c21 in poll () at /usr/lib/libc.so.6
#1  0x00007ffff72b6ee0 in  () at /usr/lib/libglib-2.0.so.0
#2  0x00007ffff72b7f62 in g_main_loop_run () at /usr/lib/libglib-2.0.so.0
#3  0x00007ffff741cc28 in  () at /usr/lib/libgio-2.0.so.0
#4  0x00007ffff72803eb in  () at /usr/lib/libglib-2.0.so.0
#5  0x00007ffff6127a9d in start_thread () at /usr/lib/libpthread.so.0
#6  0x00007ffff6d03b23 in clone () at /usr/lib/libc.so.6

Thread 4 (Thread 0x7fffdb212700 (LWP 31171)):
#0  0x00007ffff6cf8c21 in poll () at /usr/lib/libc.so.6
#1  0x00007ffff72b6ee0 in  () at /usr/lib/libglib-2.0.so.0
#2  0x00007ffff72b6fce in g_main_context_iteration () at /usr/lib/libglib-2.0.so.0
#3  0x00007ffff72b7022 in  () at /usr/lib/libglib-2.0.so.0
#4  0x00007ffff72803eb in  () at /usr/lib/libglib-2.0.so.0
#5  0x00007ffff6127a9d in start_thread () at /usr/lib/libpthread.so.0
#6  0x00007ffff6d03b23 in clone () at /usr/lib/libc.so.6

Thread 3 (Thread 0x7ffff10d8700 (LWP 31170)):
#0  0x00007ffff6cf8d16 in ppoll () at /usr/lib/libc.so.6
#1  0x00007ffff0c61a13 in pa_mainloop_poll () at /usr/lib/libpulse.so.0
#2  0x00007ffff0c61fe0 in pa_mainloop_iterate () at /usr/lib/libpulse.so.0
#3  0x00007ffff0c62091 in pa_mainloop_run () at /usr/lib/libpulse.so.0
#4  0x00007ffff71c79b0 in  () at /usr/lib/libSDL2-2.0.so.0
#5  0x00007ffff7165f9f in  () at /usr/lib/libSDL2-2.0.so.0
#6  0x00007ffff71e96ea in  () at /usr/lib/libSDL2-2.0.so.0
#7  0x00007ffff6127a9d in start_thread () at /usr/lib/libpthread.so.0
#8  0x00007ffff6d03b23 in clone () at /usr/lib/libc.so.6

Thread 2 (Thread 0x7ffff18d9700 (LWP 31169)):
#0  0x00007ffff612dafc in pthread_cond_wait@@GLIBC_2.3.2 () at /usr/lib/libpthread.so.0
#1  0x00007ffff1ec617c in  () at /usr/lib/dri/i965_dri.so
#2  0x00007ffff1ec5eb8 in  () at /usr/lib/dri/i965_dri.so
#3  0x00007ffff6127a9d in start_thread () at /usr/lib/libpthread.so.0
#4  0x00007ffff6d03b23 in clone () at /usr/lib/libc.so.6

Thread 1 (Thread 0x7ffff30a89c0 (LWP 31165)):
#0  0x0000000000000000 in  ()
#1  0x00007ffff79c4f46 in  () at /usr/lib/libgtk-3.so.0
#2  0x00007ffff79aa3f2 in  () at /usr/lib/libgtk-3.so.0
#3  0x00007ffff79d3736 in  () at /usr/lib/libgtk-3.so.0
#4  0x00007ffff79be8ab in  () at /usr/lib/libgtk-3.so.0
#5  0x00007ffff79d3670 in  () at /usr/lib/libgtk-3.so.0
#6  0x00007ffff79c0ef6 in  () at /usr/lib/libgtk-3.so.0
#7  0x00007ffff79bfc95 in  () at /usr/lib/libgtk-3.so.0
#8  0x00007ffff79c09cd in  () at /usr/lib/libgtk-3.so.0
#9  0x00007ffff79c0a1b in  () at /usr/lib/libgtk-3.so.0
#10 0x00007ffff79c0a1b in  () at /usr/lib/libgtk-3.so.0
#11 0x00007ffff79c0a1b in  () at /usr/lib/libgtk-3.so.0
#12 0x00007ffff79c0a1b in  () at /usr/lib/libgtk-3.so.0
#13 0x00007ffff79c0a1b in  () at /usr/lib/libgtk-3.so.0
#14 0x00007ffff79c0a1b in  () at /usr/lib/libgtk-3.so.0
#15 0x00007ffff79c0a1b in  () at /usr/lib/libgtk-3.so.0
#16 0x00007ffff79c0a1b in  () at /usr/lib/libgtk-3.so.0
#17 0x00007ffff79c0a1b in  () at /usr/lib/libgtk-3.so.0
#18 0x00007ffff79a69f0 in  () at /usr/lib/libgtk-3.so.0
#19 0x00007ffff739c3d5 in g_closure_invoke () at /usr/lib/libgobject-2.0.so.0
#20 0x00007ffff7389195 in  () at /usr/lib/libgobject-2.0.so.0
#21 0x00007ffff738d01e in g_signal_emit_valist () at /usr/lib/libgobject-2.0.so.0
#22 0x00007ffff738da80 in g_signal_emit () at /usr/lib/libgobject-2.0.so.0
#23 0x00007ffff778231b in  () at /usr/lib/libgdk-3.so.0
#24 0x00007ffff776cb2b in  () at /usr/lib/libgdk-3.so.0
#25 0x00007ffff72b4b63 in  () at /usr/lib/libglib-2.0.so.0
#26 0x00007ffff72b5271 in g_main_context_dispatch () at /usr/lib/libglib-2.0.so.0
#27 0x00007ffff72b6f89 in  () at /usr/lib/libglib-2.0.so.0
#28 0x00007ffff72b7f62 in g_main_loop_run () at /usr/lib/libglib-2.0.so.0
#29 0x00007ffff7a7a1af in gtk_main () at /usr/lib/libgtk-3.so.0
#30 0x000055555560620d in uiMain () at /tmp/melonDS/src/libui_sdl/libui/unix/main.c:53
#31 0x000055555559f089 in main(int, char**) (argc=1, argv=0x7fffffffe1a8) at /tmp/melonDS/src/libui_sdl/main.cpp:1592
```

**Comment by Zowayix** [comment]
Fixed by commit 65c27c5 yeehaw
